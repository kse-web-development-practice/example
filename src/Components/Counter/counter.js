import React from 'react'

import classnames from 'classnames'
import * as styles from './counter.module.css'
import PropTypes from 'prop-types'

export const Counter = ({ value, type, active }) => {
  return (
    <div
      className={classnames(styles.counterBlock, {
        [styles.counterBlockActive]: active
      })}
    >
      <span
        className={classnames(styles.counterLabel, {
          [styles.counterLabelActive]: active
        })}
      >
        Заявок
      </span>
      <span
        className={classnames(styles.counterCount, {
          [styles.counterCountTotal]: type === 'total'
        })}
      >
        {value}
      </span>
    </div>
  )
}

Counter.propTypes = {
  type: PropTypes.oneOf(['total', 'new']).isRequired,
  active: PropTypes.bool.isRequired,
  value: PropTypes.number.isRequired
}
