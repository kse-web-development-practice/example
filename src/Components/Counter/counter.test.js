import React from 'react'

import renderer from 'react-test-renderer'
import { Counter } from './counter'

describe('Counter', () => {
  it('should render', () => {
    const counterComponent = renderer.create(<Counter active={true} type="done" value={12} />)
    expect(counterComponent).toMatchSnapshot()
  })
})
