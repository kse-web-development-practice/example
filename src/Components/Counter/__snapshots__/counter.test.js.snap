// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter should render 1`] = `
<div
  className="undefined"
>
  <span
    className="undefined"
  >
    Вирішено
  </span>
  <span
    className=""
  >
    12
  </span>
</div>
`;

exports[`Counter should work 1`] = `
<div
  className="counterBlock counterBlockActive"
>
  <span
    className="counterLabel counterLabelActive"
  >
    Заявок
  </span>
  <span
    className="counterCount counterCountTotal"
  >
    12
  </span>
</div>
`;
